app/controllers/incomeSources/manage/ManageObligationsController.scala

```diff
@@ -203,18 +206,21 @@ class ManageObligationsController @Inject()(val checkSessionTimeout: SessionTime
     }
   }
 
-  def getIncomeSourceId(mode: IncomeSourceType, id: String)(implicit user: MtdItUser[_]): Either[Throwable, String] = {
+  def getIncomeSourceId(mode: IncomeSourceType, id: String, isAgent: Boolean)(implicit user: MtdItUser[_]): Either[Throwable, String] = {
+
     mode match {
       case SelfEmployment => Right(id)
       case UkProperty =>
-        user.incomeSources.properties.find(x => x.isUkProperty).map(_.incomeSourceId) match {
-          case Some(incomeSourceId) => Right(incomeSourceId)
-          case None => Left(new Error("Failed to find incomeSource Id"))
+        incomeSourceDetailsService.getActiveUkPropertyFromUserIncomeSources match {
+          case Right(ukProperty: PropertyDetailsModel) => Right(ukProperty.incomeSourceId)
+          case Left(error: Error) => Left(error)
+          case _ => Left(new Error("Unknown error"))
```

Comment:
please provide more details in the error, ie. incomeSourceId and IncomeSourceType